### ISSUE TITLE
Spawn logic could be improved when recovering from an economic crisis :)

### ISSUE THREAD

**Issue by schoolie** [schoolie3344/ScreepsCoop#29]
Current spawn logic always tries to build the biggest creeps the room can handle, unless there are no creeps left at all, then it spawns a tiny one.

**Comment by schoolie3344** [comment]
Yeah, I did a little bit with this a while back and scrapped it. I guess I
was planning on avoiding genocide for the most part ;)
With all the modifications being added, though, I think it would be a good
idea. I think bodyPartCap or whatever it was could be modified by a creep
count check before the .push methods create the bodyParts array. If 0
creeps make 300 cost creep, if < 3 creeps maybe a 2,2,2 creep, and so on.

**Comment by schoolie** [comment]
One idea: 
Determine the max energy harvest rate based on the number of WORK parts on creeps in the room. Base the size of new creeps on that number, targeting the ability be spawning creeps X% of the time (25% ?) based on how fast the existing creeps can refill the stores.

**Comment by schoolie** [comment]
Meeting in the middle between your idea and mine: 

Rather than counting creeps, count WORK parts. 
0-2 WORK parts: bodyPartCap = 3
3-5 WORK parts: bop
etc...
